music21/musicxml/m21ToXml.py

```diff
@@ -2717,7 +2766,7 @@ def instrumentToXmlScoreInstrument(self, i):
         </score-instrument>
         '''
         mxScoreInstrument = Element('score-instrument')
-        mxScoreInstrument.set('id', str(i.instrumentId))
```

Comment:
I don't understand this removal.  It just makes something break that has been working in the past in case someone sets the id to a number or None.

Reply:
Happy to remove. I think it was because `.set()` handles that--in fact, better in case of None, see:

```
>>> mxArt = EL('<spiccato placement="above"/>')
>>> mxArt.set('lucky_number', 3.14159)
>>> mxArt.get('lucky_number')
3.14159
>>> mxArt.set('lucky_number', None)
>>> mxArt.get('lucky_number') is None
True
>>> mxArt.set('lucky_number', str(None))
>>> mxArt.get('lucky_number')
'None'
```

Reply:
Actually, the None case may have been why I did this edit, I can't remember at the moment.

Reply:
Supe.  I did not know that `set` took an any.

Reply:
Yeah, I think the risk of "None" being written out is worth the change.

Reply:
Nononono. I'll revert this. Better to write out "None" than to crash:

```
        >>> i.instrumentId = None
        >>> PEX = musicxml.m21ToXml.PartExporter()
        >>> mxMidiInstrument = PEX.instrumentToXmlMidiInstrument(i)
        >>> PEX.dump(mxMidiInstrument)
```
```
----------------------------------------------------------------------
File "/Users/jwalls/music21/music21/musicxml/m21ToXml.py", line 2847, in __main__.PartExporter.instrumentToXmlMidiInstrument
Failed example:
    PEX.dump(mxMidiInstrument)
Exception raised:
    Traceback (most recent call last):
      File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/xml/etree/ElementTree.py", line 1052, in _escape_attrib
        if "&" in text:
    TypeError: argument of type 'NoneType' is not iterable

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest __main__.PartExporter.instrumentToXmlMidiInstrument[9]>", line 1, in <module>
        PEX.dump(mxMidiInstrument)
      File "/Users/jwalls/music21/music21/musicxml/m21ToXml.py", line 791, in dump
        return helpers.dump(obj)
      File "/Users/jwalls/music21/music21/musicxml/helpers.py", line 69, in dump
        print(dumpString(obj))
      File "/Users/jwalls/music21/music21/musicxml/helpers.py", line 46, in dumpString
        xStr = ET.tostring(xmlEl, encoding='unicode')
      File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/xml/etree/ElementTree.py", line 1109, in tostring
        ElementTree(element).write(stream, encoding,
      File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/xml/etree/ElementTree.py", line 748, in write
        serialize(write, self._root, qnames, namespaces,
      File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/xml/etree/ElementTree.py", line 906, in _serialize_xml
        v = _escape_attrib(v)
      File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/xml/etree/ElementTree.py", line 1075, in _escape_attrib
        _raise_serialization_error(text)
      File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/xml/etree/ElementTree.py", line 1029, in _raise_serialization_error
        raise TypeError(
    TypeError: cannot serialize None (type NoneType)
```

Reply:
I think the changes in this PR already randomize when we encounter None, but adding the safety back is good.